app/controllers/works_controller.rb

```diff
@@ -12,6 +12,10 @@ def root
 
   def index
     @works_by_category = Work.to_category_hash
+    if @user.nil?
+      flash[:result_text] = "You must log in to do that"
+      return redirect_to root_path
+    end
```

Comment:
Instead of repeating this in several methods, it's better to use a controller filter.